nsqd/nsqd.go

```diff
@@ -223,6 +239,27 @@ func (n *NSQD) Main() {
 	if n.opts.StatsdAddress != "" {
 		n.waitGroup.Wrap(func() { n.statsdLoop() })
 	}
+
+	var httpsAddr *net.TCPAddr
+	if n.httpsListener != nil {
+		httpsAddr = n.httpsListener.Addr().(*net.TCPAddr)
+	}
+
+	if n.opts.GossipAddress != "" {
+		serf, err := initSerf(n.opts, n.serfEventChan,
+			n.tcpListener.Addr().(*net.TCPAddr),
+			n.httpListener.Addr().(*net.TCPAddr),
+			httpsAddr,
+			n.broadcastAddr)
+		if err != nil {
+			n.logf("FATAL: failed to initialize Serf - %s", err)
+			os.Exit(1)
+		}
+		n.serf = serf
+	}
+
+	n.waitGroup.Wrap(func() { n.serfEventLoop() })
+	n.waitGroup.Wrap(func() { n.gossipLoop() })
```

Comment:
yes, they should be inside that block and remove the escape logic within the funcs (I prefer doing it externally)